### ISSUE TITLE
exception raised on update with a join in the where clause

### ISSUE THREAD

**Issue by backwardspy** [piccolo-orm/piccolo#625]
i couldn't think of a concise way to explain my problem so here's an example:

```python
In [19]: await Band.select(Band.name).where(Band.manager.name == 'Guido')
SELECT "band"."name" FROM band LEFT JOIN manager band$manager ON (band.manager = band$manager.id) WHERE "band$manager"."name" = 'Guido'
Out[19]: [{'name': 'Pythonistas'}]

In [20]: await Band.update({Band.name: 'The Pythonistas'}).where(Band.manager.name == 'Guido')
UPDATE band SET "name" = 'The Pythonistas' WHERE "band$manager"."name" = 'Guido'
---------------------------------------------------------------------------
OperationalError                          Traceback (most recent call last)
Cell In [20], line 1
----> 1 await Band.update({Band.name: 'The Pythonistas'}).where(Band.manager.name == 'Guido')

... traceback removed ...

OperationalError: no such column: band$manager.name
```

**Comment by backwardspy** [comment]
okay cool yeah i figured i'd go with something like that for now. if i get time later i may look into what it'd take to implement this kind of update query in piccolo, unless you're already working on it.

**Comment by dantownsend** [comment]
I haven't looked much into it yet. Just enough to know it's possible.

**Comment by dantownsend** [comment]
Fixed in `0.106.0`.
